src/Sylius/Bundle/ApiBundle/CommandHandler/RegisterShopUserHandler.php

```diff
@@ -66,15 +68,17 @@ public function __invoke(RegisterShopUser $command): void
         $customer->setPhoneNumber($command->phoneNumber);
         $customer->setUser($user);
 
-        $this->handleVerification($user);
+        $this->handleVerificationInChannel($user, $command->channelCode);
 
         $this->shopUserManager->persist($user);
     }
 
-    private function handleVerification(ShopUserInterface $user): void
+    private function handleVerificationInChannel(ShopUserInterface $user, ?string $channelCode): void
```

Comment:
```suggestion
    private function handleVerificationInChannel(ShopUserInterface $user, string $channelCode): void
```
and the assertion could be before calling this method